File: client/src/app/models/verb.ts
import {BaseTense} from "./base-tense";

export class Verb {

  // infinitive: string;
  // presentTense: string[];
  // presentPerfectTense: string[];
  tenseArray: Array<BaseTense>;
  presentName = "present-tense";
  presentPerfectName = "present-perfect-tense";

  constructor(
    infinitive: string,
    presentTenseList: string[],
    presentPerfectTenseList: string[],
    imperfectTenseList: string[],
    futureTenseList: string[]
  ) {
    console.log(infinitive);
    console.log(presentTenseList);
    this.tenseArray = new Array<BaseTense>();

    for (let index in presentTenseList) {
      this.tenseArray.push({
        present: presentTenseList[index],
        presentPerfect: presentPerfectTenseList[index],
        imperfect: imperfectTenseList[index],
        future: futureTenseList[index]
      });

      console.log(this.tenseArray[index]);
    }
  }
}

File: server/src/main/java/dev/aubique/conj/models/Verb.java
package dev.aubique.conj.models;

import java.util.List;

public class Verb {

    private String infinitive;
    private String presentName = "Present Simple Tense";
    private List<String> presentTense;
    private String presentPerfectName = "Present Perfect Tense";
    private List<String> presentPerfectTense;

    public Verb(String infinitive, List<String> presentTense, List<String> presentPerfectTense) {
        this.infinitive = infinitive;
        this.presentTense = presentTense;
        this.presentPerfectTense = presentPerfectTense;
    }

    public String getInfinitive() {
        return infinitive;
    }

    public String getPresentName() {
        return presentName;
    }

    public List<String> getPresentTense() {
        return presentTense;
    }

    public String getPresentPerfectName() {
        return presentPerfectName;
    }

    public List<String> getPresentPerfectTense() {
        return presentPerfectTense;
    }
}
